src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -153,8 +166,348 @@ public User updateUser(User user) {
+    otherUser.removeLikeFromUser(user);
+
+    // then create a new match between users
+    Match match = new Match();
+    // Yes, it would be absolutely possible to do without the Pair class...
+    Pair<User, User> pair = new Pair<>(user, otherUser);
+    match.setUserPair(pair);
+    matchRepository.saveAndFlush(match);
+
+    return match;
+  }
+
+  /**
+   * Deletes any game from User that is not active (which would be stupid to do)
+   * @param user: User from whom the game shall be deleted from
+   * @param game: Game which shall be deleted
+   */
+  public void deleteGameIfEmpty(User user, Game game) {
+    // if game is not active and has no more plays, we can delete it.
+    if(game.getPlays().isEmpty() && game.getGameStatus() != GameStatus.ACTIVE) {
+      user.deletePastGame(game);
+    }
+    userRepository.saveAndFlush(user);
+  }
+
+  /**
+   * Move Active Game to Past Games IF the time is up
+   * @param userId: userId from User
+   */
+  public void updateActiveGameIfNecessary(Long userId) {
+      updateActiveGameIfNecessary(getUserById(userId));
+  }
+
+  /**
+   * Move Active Game to Past Games IF the time is up
+   * @param user: User from whom the active games shall be updated
+   */
+  private void updateActiveGameIfNecessary(User user) {
+    Game activeGame = user.getActiveGame();
+    // case there is no active game, just return
+    if (activeGame == null) return;
+
+    // calculate how old the active game is
+    long diffTime = new Date().getTime() - activeGame.getCreationTime().getTime();
+    // case the game is older than one Day, put it to the past games
+    if(diffTime > Time.ONE_DAY) {
+      // update the user who was a candidate for black cards
+      activeGame.setGameStatus(GameStatus.INACTIVE);
+      user.flushGameToPastGames();
+      // TODO: 12.04.2022 SaveAndFlush GameRepository here? (IDE doesn't complain until now)
+      userRepository.saveAndFlush(user);
+    }
+    // case the active game is not older than 24 hours, just return
+  }
+
+  /**
+   * Get the current black cards of a user: automatically delete old black cards that are older than some time period
+   * @param userId: userId from the user from whom we want to get the current black cards
+   * @return: a list of black cards - can be empty
+   */
+  public List<BlackCard> getCurrentBlackCards(Long userId) {
+    User user = getUserById(userId);
+    // get current black cards
+    UserCards currentCards = user.getUserCards();
+    // return empty list if user has no current black cards (last two conditions are a safety net)
+    if (currentCards == null || currentCards.getBlackCards() == null || currentCards.getBlackCards().isEmpty()) {
+      return Collections.emptyList();
+    }
+    // return current black cards if they're younger than one day, else return empty list
+    long diffTime = new Date().getTime() - currentCards.getDate().getTime();
+    if(diffTime > Time.ONE_DAY) {
+      user.setUserCards(null);
```

Comment:
Also deletes users white cards?
What if user is playing (white cards) while this occures? (Is that even possible?)

Reply:
I splitted that class in half, the white cards are now independet of the black cards, and in no case get set to null, but are only overwritten.